diff --git a/node_modules/prisma-nestjs-graphql/index.js b/node_modules/prisma-nestjs-graphql/index.js
index b2165bb..0b23d58 100644
--- a/node_modules/prisma-nestjs-graphql/index.js
+++ b/node_modules/prisma-nestjs-graphql/index.js
@@ -322,7 +322,9 @@ var ImportDeclarationMap = class extends Map {
         }
       ];
     }
-    this.add(name, value);
+    if (value.moduleSpecifier !== '') {
+      this.add(name, value);
+    }
   }
   *toStatements() {
     const iterator = this.values();
@@ -655,7 +657,7 @@ function getPropertyType(args) {
     }
     case "Json": {
       return [
-        "any"
+        "object"
       ];
     }
     case "Null": {
